### PR TITLE
[NCL-8251][NCL-8389] clear endpoint + cancel timeout with failover

### PR DESCRIPTION
**Pull request by michalovjan**
Hi @thescouser89. There is a lot in this PR so I'd suggest to go commit by commit.

In general (smaller changes):
- update to Quarkus 3.x (with backward compatible `Javax` classified packages)
- extraction of application configuration into CDI Bean/object/interface (ApplicationConfig.class)
- test refactor (shared AbstractClass, repackaging)
- tests now do not output everything into STDOUT but into files in target/surefire-reports
- Jeager docker-compose.yaml for looking at OTEL from tests or dev mode
- [NCL-8389] /clear-all maintenance endpoint for cases when we have stuck 
- OIDC accessToken now refreshes on each request which includes ones that are created with Retries
- Fixed ISPN classpath pollution
- and more stuff

Timeout job
- the timeout can be configured on per-task basis
- It is a clustered job which means that it is registered in the cluster and failed-over in case node shutdowns. A clustered job is owned so every instance has to have unique ID to differentiate (I've used `quarkus.uuid` in `scheduler.name`)
- it's independent and can run at any time at any instance

Failover
- the failover mechanism is currently triggered only on Shutdown of a instance (will implement more triggers in next issue)
- In simple terms the failover mechanism works by: 
1. query all 'local' resources. 
2. For each resource, put it into signal cache
3. Every Rex instance has a Listener to changes on signal cache so 2. step can be thought of a broadcast
4. Every instance will create a Transaction and tries to remove the resource from signal cache. Only one can succeed so only one instance takesover the resource.
5. The instance that won, will instantiate the resource (in this case the TimeourCancelClusteredJob) and runs it.

### CODE REVIEW

**Comment by michalovjan** [comment]
retest this please
